consumergroup/offset_manager.go

```diff
@@ -0,0 +1,155 @@
+package consumergroup
+
+import (
+	"sync"
+	"time"
+)
+
+type OffsetManager interface {
+	InitializePartition(topic string, partition int32) (int64, error)
+	MarkAsSeen(topic string, partition int32, offset int64) bool
+	Close() error
+}
+
+type OffsetManagerConfig struct {
+	CommitInterval time.Duration
+	VerboseLogging bool
+}
+
+type ZookeeperOffsetManager struct {
```

Comment:
this doesn't need to be public if the constructor is just returning the interface